codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/testutil/CodegenIntegrationTest.kt

```diff
@@ -68,15 +66,15 @@ abstract class DecoratableBuildPlugin<T, C : CodegenContext> : SmithyBuildPlugin
 }
 
 // TODO(https://github.com/awslabs/smithy-rs/issues/1864): move to core once CodegenDecorator is in core
-private fun <T, C : CodegenContext> codegenIntegrationTest(
+private fun codegenIntegrationTest(
     model: Model,
-    buildPlugin: DecoratableBuildPlugin<T, C>,
-    additionalDecorators: List<RustCodegenDecorator<T, C>>,
+    buildPlugin: DecoratableBuildPlugin,
+    additionalDecorators: List<ClientCodegenDecorator>,
```

Comment:
this should be a generic so that this function can also be used by the server if possible

Reply:
It doesn't look like `codegen-server` uses this at all. I'm more inclined to move it into the `test/` directory and remove the TODO comment about moving the decorator into core.

Reply:
Looks like the `sdk-codegen` module uses it, so it needs to stay in `src/`.